route-rs-afpacket/tests/integration_test.rs

```diff
@@ -0,0 +1,168 @@
+#![cfg(target_os = "linux")]
+
+use afpacket;
+use rand::{self, Rng};
+use route_rs_afpacket::{AfPacketInputLink, AfPacketOutputLink};
+use route_rs_packets as packets;
+use route_rs_runtime::{
+    link::LinkBuilder,
+    utils::test::{harness::run_link, packet_generators::immediate_stream},
+};
+use std::{ffi::CString, net, time::Duration};
+use tokio::{self, runtime, time};
+
+fn create_socket() -> (afpacket::SendHalf, afpacket::RecvHalf) {
+    let iface = {
+        let iface = CString::new("lo");
+        assert!(iface.is_ok());
+        iface.unwrap()
+    };
+    let sock = {
+        let sock = afpacket::AsyncBoundSocket::from_interface(&iface);
+        assert!(sock.is_ok());
+        sock.unwrap()
+    };
+
+    sock.split()
+}
+
+fn make_packet(rng: &mut impl Rng) -> Vec<u8> {
+    let body = {
+        let mut body = vec![0; 64];
+        rng.fill(&mut body[..]);
+        body
+    };
+    let mut udp_pkt = packets::UdpSegment::empty();
+    udp_pkt.set_src_port(3001);
+    udp_pkt.set_dest_port(3002);
+    udp_pkt.set_payload(&body);
+    let mut ipv4_pkt = packets::Ipv4Packet::encap_udp(udp_pkt);
+    ipv4_pkt.set_src_addr(net::Ipv4Addr::new(10, 0, 0, 1));
+    ipv4_pkt.set_dest_addr(net::Ipv4Addr::new(10, 0, 0, 2));
+    ipv4_pkt.set_ttl(2);
+    let mut eth_pkt = packets::EthernetFrame::encap_ipv4(ipv4_pkt);
+    eth_pkt.set_src_mac(packets::MacAddr::new([0xff, 0xff, 0xff, 0xff, 0xff, 0xff]));
+    eth_pkt.set_dest_mac(packets::MacAddr::new([0xff, 0xff, 0xff, 0xff, 0xff, 0xff]));
+
+    eth_pkt.data
+}
+
+#[test]
+#[should_panic]
+#[ignore]
+fn output_panics_when_built_without_ingressor() {
+    let (s, _) = create_socket();
+
+    AfPacketOutputLink::new().channel(s).build_link();
+}
+
+#[test]
+#[should_panic]
+fn output_panics_when_built_without_channel() {
+    let packet_generator = immediate_stream(vec![]);
+
+    AfPacketOutputLink::new()
+        .ingressor(packet_generator)
+        .build_link();
+}
+
+#[test]
+#[should_panic]
+#[ignore]
+fn output_panics_when_built_with_multiple_ingressors() {
+    let (s, _) = create_socket();
+    let pkt_gen_1 = immediate_stream(vec![]);
+    let pkt_gen_2 = immediate_stream(vec![]);
+
+    AfPacketOutputLink::new()
+        .ingressors(vec![pkt_gen_1, pkt_gen_2])
+        .channel(s)
+        .build_link();
+}
+
+#[test]
+#[should_panic]
+fn input_panics_when_built_with_ingressors() {
+    AfPacketInputLink::new()
+        .ingressors(vec![immediate_stream(vec![])])
+        .build_link();
+}
+
+#[test]
+#[should_panic]
+fn input_panics_when_built_without_channel() {
+    AfPacketInputLink::new().build_link();
+}
+
+#[test]
+#[ignore]
+fn input_identity() {
```

Comment:
Why are these tests ignored? If they can't be run by the CI can you leave a small comment about how to run them locally?